pkg/services/annotations/annotationsimpl/composite_store.go

```diff
@@ -0,0 +1,98 @@
+package annotationsimpl
+
+import (
+	"context"
+	"errors"
+	"slices"
+	"sync"
+
+	"github.com/grafana/grafana/pkg/services/annotations"
+	"github.com/grafana/grafana/pkg/services/annotations/accesscontrol"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+type CompositeStore struct {
+	primary   store
+	historian store
```

Comment:
We do a lot of special stuff with `primary`, i.e. only dispatching writes to it. We could consider naming them `writer` and `reader` instead, so it's clear immediately what the difference is? Also I think it'd be good to drop the `historian` name, as it feels like a leak of the abstraction.

Reply:
Implemented this as multiple readers, rather than a `writer`/`reader` since the `writer` in this case would also be handling read actions.